src/otx/cli/test.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""CLI entrypoint for testing."""
+# ruff: noqa
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from hydra import compose, initialize
+from jsonargparse import ArgumentParser
+from hydra.core.global_hydra import GlobalHydra
+
+from otx.cli.utils.hydra import configure_hydra_outputs
+
+if TYPE_CHECKING:
+    from jsonargparse._actions import _ActionSubCommands
+
+def add_test_parser(subcommands_action: _ActionSubCommands) -> None:
+    """Add subparser for test command.
+
+    Args:
+        subcommands_action (_ActionSubCommands): Sub-Command in CLI.
+
+    Returns:
+        None
+    """
+    parser = ArgumentParser()
+    parser.add_argument("overrides", help="overrides values", default=[], nargs="+")
+    subcommands_action.add_subcommand("test", parser, help="Testing subcommand for OTX")
+
+
+def otx_test(overrides: list[str]) -> None:
+    """Main entry point for testing.
+
+    :param overrides: Override List values.
+    :return: Optional[float] with optimized metric value.
+    """
+    initialize(config_path="../config", version_base="1.3", job_name="otx_test")
+    cfg = compose(config_name="test", overrides=overrides, return_hydra_config=True)
+    configure_hydra_outputs(cfg)
+
+    # test the model
+    from otx.core.engine.test import test
+    metric_dict, _ = test(cfg)
+
+    # Clear the Hydra instances.
+    GlobalHydra().clear()
```

Comment:
`initialize` seems Python context manager. Please see https://github.com/facebookresearch/hydra/blob/2f7628756b953fa35d8f7a91fec1018a55cac43b/hydra/initialize.py#L98-L102.
Can we replace this with Python `with` statement? If possible, we must apply it `train.py` as well.

Reply:
Thanks, I'll try

Reply:
This seems to work fine. Thanks. - https://github.com/openvinotoolkit/training_extensions/pull/2707/commits/c5ac6a30b96c5400a1c9f88d3f4d2ca5e50ea2e5